### PR TITLE
cliccl/debug_backup.go: refactor backup inspection tool

### PR DESCRIPTION
**Pull request by Elliebababa**
Previously, cli backup inspection tool was incorporated
in cliccl/load.go. Both the command and syntax are
confusing in terms of backup inspection.

This patch refactors the naming and syntax of the tool
and hang the tool under `cockroach debug`, so that
it makes more sense when using it to debug backups.

Release note (cli change): Backup inspection used to be done via
`cockroach load show ..`, which was confusing to users with
ambiguous verbs in the command chain. We refactor the syntax of
tool and make the syntax  more clear and indicative for users to
debug backups. The changes are:
`load show summary <backup_url>` ->
`debug backup show <backup_url>`
```
$ ./cockroach debug backup show $AWS_BACKUP_PATH
{
	"StartTime": "1970-01-01T00:00:00Z",
	"EndTime": "2021-04-05T22:02:25Z",
	"DataSize": "110 KiB",
	"Rows": 224,
	"IndexEntries": 324,
	"FormatVersion": 1,
	"ClusterID": "7c1b9ceb-d05a-4879-b4a4-886ace456953",
	"NodeID": 0,
	"BuildInfo": "CockroachDB CCL v21.1.0-alpha.3-1790-g25dc628337-dirty (x86_64-apple-darwin19.6.0, built 2021/03/29 18:50:47, go1.15.7)",
	"Files": [
		{
			"Path": "647462079350243329.sst",
			"Span": "/Table/4/{1-2}",
			"DataSize": "99 B",
			"IndexEntries": 0,
			"Rows": 2
		},
	],
	"Spans": "[/Table/4/{1-2}]",
	"DatabaseDescriptors": {
		"1": "system"
	},
	"TableDescriptors": {
		"53": "demodb.public.demo",
		"54": "demodb.public.foo"
	},
	"TypeDescriptors": {},
	"SchemaDescriptors": {
		"29": "public"
	}
}
```

`load show incremental <backup_url>` ->
`debug backup list-incremental <backup_url>`
```
$ ./cockroach debug backup list-incremental $AWS_BACKUP_PATH
              path             |      start time      |       end time
-------------------------------+----------------------+-----------------------
  /mybackup                    | -                    | 2021-04-09T20:22:59Z
  /mybackup/20210409/202303.20 | 2021-04-09T20:22:59Z | 2021-04-09T20:23:03Z
  /mybackup/20210409/202305.97 | 2021-04-09T20:23:03Z | 2021-04-09T20:23:05Z
(3 rows)
```

`load show backups <collection_url>` ->
`debug backup list-backups <collection_url>`
```
$ ./cockroach debug backup list-backups $AWS_COLLECTION_PATH
           path
--------------------------
  ./2021/04/09-202455.98
  ./2021/04/09-202522.59
  ./2021/04/09-202526.93
(3 rows)
```

`load show data <table_name> <backup_url> `->
`debug backup export <backup_url> --table=<table_name> [--destination=</path/to/export>]` 
```
$ ./cockroach debug backup export $AWS_BACKUP_PATH --table=demodb.public.demo
1,null,true
2,'ellie',null
```

### CODE REVIEW

File: pkg/ccl/cliccl/debug_backup.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2017 The Cockroach Authors.
+// Copyright 2021 The Cockroach Authors.
```
Comment (pbardea) [inline pkg/ccl/cliccl/debug_backup.go:1]: I don't think we usually change the copyright date on file renames.

**Review by pbardea — APPROVED** [review]

File: pkg/ccl/cliccl/debug_backup.go (hunk 1 above)
Comment (Elliebababa) [reply pkg/ccl/cliccl/debug_backup.go]: oops. done.

File: pkg/ccl/backupccl/targets.go [hunk 2]
```diff
@@ -472,15 +473,17 @@ func MakeBackupTableEntry(
 		ind := -1
 		for i, b := range backupManifests {
 			if b.StartTime.Less(endTime) && endTime.LessEq(b.EndTime) {
-				if endTime != b.EndTime {
-					return BackupTableEntry{}, errors.Newf("should specify the exact backup time, next backup timestamp is %s", b.EndTime.AsOfSystemTime())
+				if endTime != b.EndTime && b.MVCCFilter != MVCCFilter_All {
+					return BackupTableEntry{}, errors.Newf(
```
Comment (knz) [inline pkg/ccl/backupccl/targets.go:477]: nit: you should use an error hint here and keep the main message short.

See: https://wiki.crdb.io/wiki/spaces/CRDB/pages/1676804870/Error+concepts+and+handling+in+CockroachDB#Error-messages%2C-hints-and-codes

File: pkg/sql/row/kv_fetcher.go [hunk 3]
```diff
@@ -210,9 +210,15 @@ func (f *BackupSSTKVFetcher) nextBatch(
 			}
 		}
 		if len(f.iter.UnsafeValue()) == 0 {
-			// Value is deleted.
-			f.iter.NextKey()
-			continue
+			if f.endTime.IsEmpty() || f.iter.UnsafeKey().Timestamp.Less(f.endTime) {
```
Comment (knz) [inline pkg/sql/row/kv_fetcher.go:213]: This change either:

- does not belong to this PR, or
- needs a unit test, AND a review from the KV and SQL queries team.

**Review by knz — CHANGES_REQUESTED** [review]

**Review by pbardea — APPROVED** [review]

**Comment by Elliebababa** [comment]
bors r=pbardea

File: pkg/ccl/backupccl/targets.go (hunk 2 above)
Comment (Elliebababa) [reply pkg/ccl/backupccl/targets.go]: Thanks for catching! (I just found github ate my comments somehow

File: pkg/sql/row/kv_fetcher.go (hunk 3 above)
Comment (Elliebababa) [reply pkg/sql/row/kv_fetcher.go]: Yep, this change does not belong to this PR. Backup related tests were in ccl pkg. Not sure if we can import ccl from sql pkg. So I made the unit tests in cliccl/debug_backup_test.go.
